### PR TITLE
Make authentication provider discovery AOT-safe

### LINKED ISSUE #4193: Entra ID authentication broken under NativeAOT in v7.0 — Extensions.Azure reflection-based discovery is AOT-incompatible
### Describe the bug

SqlClient 7.0 moved `ActiveDirectoryAuthenticationProvider` from the core `Microsoft.Data.SqlClient` assembly into the separate `Microsoft.Data.SqlClient.Extensions.Azure` package. The provider is now discovered at runtime via reflection (`Assembly.Load` + `Activator.CreateInstance`), which breaks under NativeAOT because the linker has no static reference to follow and trims the assembly.

In v6.x, the provider was part of the core assembly and registered via a direct `new ActiveDirectoryAuthenticationProvider(...)` call in `SqlAuthenticationProviderManager.SetDefaultAuthProviders()`. This worked correctly under NativeAOT because the AOT compiler could trace the entire type hierarchy statically.

In v7.0, `SqlAuthenticationProviderManager`'s static constructor ([source](https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlAuthenticationProviderManager.cs)) uses:
```csharp
var assembly = Assembly.Load("Microsoft.Data.SqlClient.Extensions.Azure");
var type = assembly.GetType("Microsoft.Data.SqlClient.ActiveDirectoryAuthenticationProvider");
var instance = Activator.CreateInstance(type, ...) as SqlAuthenticationProvider;
```

Under NativeAOT, this fails silently (caught by the exception filter), leaving the `_providers` dictionary empty. When a connection with `Authentication=Active Directory Managed Identity` (or any other Entra ID method) is opened, `GetFedAuthToken` calls `SqlAuthenticationProviderManager.GetProvider()`, gets `null`, and throws `SqlException: Cannot find an authentication provider for 'ActiveDirectoryManagedIdentity'`.

**Calling `SqlAuthenticationProvider.SetProvider()` from application code also does not work**, because the public `SetProvider` API in `Microsoft.Data.SqlClient.Extensions.Abstractions` communicates with the internal `SqlAuthenticationProviderManager` via reflection as well ([source](https://github.com/dotnet/SqlClient/blob/main/src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/SqlAuthenticationProvider.Internal.cs)):
```csharp
var assembly = Assembly.Load("Microsoft.Data.SqlClient");
var manager = assembly.GetType("Microsoft.Data.SqlClient.SqlAuthenticationProviderManager");
_setProvider = manager.GetMethod("SetProvider", BindingFlags.NonPublic | BindingFlags.Static);
// ...
_setProvider.Invoke(null, [authenticationMethod, provider]);
```

This means there is **no AOT-safe way to register an authentication provider** in v7.0.

None of these code paths have `[RequiresUnreferencedCode]`, `[RequiresDynamicCode]`, or `[DynamicallyAccessedMembers]` annotations, so the AOT compiler emits no warnings — the failure is entirely silent until runtime.

```
Exception message:
Microsoft.Data.SqlClient.SqlException (0x80131904):
  Cannot find an authentication provider for 'ActiveDirectoryManagedIdentity'.

Stack trace:
   at Microsoft.Data.SqlClient.Connection.SqlConnectionInternal.GetFedAuthToken(SqlFedAuthInfo fedAuthInfo)
   at Microsoft.Data.SqlClient.Connection.SqlConnectionInternal.OnFedAuthInfo(SqlFedAuthInfo fedAuthInfo)
   at Microsoft.Data.SqlClient.TdsParser.TryRun(RunBehavior runBehavior, ...)
   at Microsoft.Data.SqlClient.TdsParser.Run(RunBehavior runBehavior, ...)
   at Microsoft.Data.SqlClient.Connection.SqlConnectionInternal.CompleteLogin(Boolean enlistOK)
   at Microsoft.Data.SqlClient.Connection.SqlConnectionInternal.AttemptOneLogin(...)
   at Microsoft.Data.SqlClient.Connection.SqlConnectionInternal.LoginNoFailover(...)
   at Microsoft.Data.SqlClient.Connection.SqlConnectionInternal.OpenLoginEnlist(...)
   at Microsoft.Data.SqlClient.Connection.SqlConnectionInternal..ctor(...)
```

### To reproduce

Minimal reproduction — a NativeAOT-published app connecting to Azure SQL with Entra ID Managed Identity:

```csharp
// Program.cs
using Microsoft.Data.SqlClient;

var connStr = "Server=myserver.database.windows.net;Database=mydb;Authentication=Active Directory Managed Identity;";

using var connection = new SqlConnection(connStr);
await connection.OpenAsync(); // throws: Cannot find an authentication provider for 'ActiveDirectoryManagedIdentity'
```

Project file:
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net10.0</TargetFramework>
    <PublishAot>true</PublishAot>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="Microsoft.Data.SqlClient" Version="7.0.0" />
    <PackageReference Include="Microsoft.Data.SqlClient.Extensions.Azure" Version="1.0.0" />
  </ItemGroup>
</Project>
```

Publish and run:
```bash
dotnet publish -r linux-x64 -c Release
./bin/Release/net10.0/linux-x64/publish/MyApp
```

This works correctly with `Microsoft.Data.SqlClient` 6.1.4 (no Extensions.Azure package needed), because `ActiveDirectoryAuthenticationProvider` was in the core assembly with a direct `new` instantiation.

### Expected behavior

Entra ID authentication methods (Managed Identity, Workload Identity, Default, etc.) should work under NativeAOT, as they did in v6.x.

At minimum, the reflection-based code paths should be annotated with `[RequiresUnreferencedCode]` / `[RequiresDynamicCode]` so that the AOT compiler warns users that these code paths are unsafe.

Ideally, an AOT-compatible registration path should be provided — for example:
- A direct (non-reflection) `SetProvider` overload on `SqlAuthenticationProviderManager` accessible from application code
- Or a builder/options pattern that lets users register the provider at compile time

### Workaround

The `AccessTokenCallback` property on `SqlConnection` bypasses the `SqlAuthenticationProvider` registry entirely and works under NativeAOT. Combined with `Azure.Identity` (which is AOT-compatible), this is currently the only reliable approach:

```csharp
var builder = new SqlConnectionStringBuilder(connectionString);
var credential = new ManagedIdentityCredential(); // from Azure.Identity

// Strip the Authentication keyword so SqlClient doesn't try the provider registry
builder.Authentication = SqlAuthenticationMethod.NotSpecified;

using var connection = new SqlConnection(builder.ConnectionString);
connection.AccessTokenCallback = async (parameters, cancellationToken) =>
{
    var token = await credential.GetTokenAsync(
        new Azure.Core.TokenRequestContext(["https://database.windows.net/.default"]),
        cancellationToken);
    return new SqlAuthenticationToken(token.Token, token.ExpiresOn);
};
await connection.OpenAsync(); // works under NativeAOT
```

### Further technical details

Microsoft.Data.SqlClient version: 7.0.0
.NET target: .NET 9.0 / .NET 10.0 (both affected)
SQL Server version: Azure SQL Database
Operating system: Linux (Ubuntu 24.04 Noble, Docker container with `mcr.microsoft.com/dotnet/runtime-deps:10.0-noble-chiseled`)

**Additional context**

- Related: #1947 (NativeAOT support: get to zero warnings)
- Related: #2742 (native aot deployment error using Microsoft Entra authentication — different root cause in v5.x, `MemoryCache` static constructor failure)
- The v6 → v7 change: in v6.1.4, `SqlAuthenticationProviderManager.SetDefaultAuthProviders()` called `new ActiveDirectoryAuthenticationProvider(...)` directly. In v7.0, this was replaced with `Assembly.Load("Microsoft.Data.SqlClient.Extensions.Azure")` + `Activator.CreateInstance`.
- `SqlAuthenticationProvider.SetProvider()` (the public API intended for manual provider registration) also fails under NativeAOT because `SqlAuthenticationProvider.Internal` class uses `MethodInfo.Invoke` with `BindingFlags.NonPublic` to reach the internal `SqlAuthenticationProviderManager`.

### PR DESCRIPTION
**Pull request by paulmedynski**
## Description

Makes SqlClient's authentication provider discovery safe for NativeAOT.

Stacked on #4670, which moves the provider registry into Abstractions and removes the Abstractions→SqlClient reflection. This PR handles the *other* reflection edge: SqlClient→Azure extension discovery.

### The problem

`AuthenticationBootstrapper` discovers `app.config` providers and the Azure extension via `Type.GetType`, `Assembly.Load`, and `Activator.CreateInstance`. Under NativeAOT the trimmer has no static reference to follow, so it trims the Azure extension and every Active Directory authentication method silently loses its provider (#4193). None of the reflection carried trim annotations, so the compiler emitted no warnings.

### The change

- **New `EnableReflectionBasedAuthenticationProviderDiscovery` app context switch** (default `true`), guarding the bootstrapper's discovery. On .NET 9+ it carries `[FeatureSwitchDefinition]`; `ILLink.Substitutions.xml` provides the same substitution on .NET 8.
- **`[RequiresUnreferencedCode]` / `[RequiresDynamicCode]`** on `LoadConfiguration` and `LoadAzureExtensionProvider`, so trim analysis warns even when the switch is left enabled.
- **`tools/AotCompatibility`** — a standalone publish harness that verifies the substitution actually removes `LoadAzureExtensionProvider` from the native image.

An AOT app opts in at publish time:

```xml
<RuntimeHostConfigurationOption
    Include="Microsoft.Data.SqlClient.EnableReflectionBasedAuthenticationProviderDiscovery"
    Value="false"
    Trim="true" />
```

The trimmer folds the guard to a constant `false`, eliminates the dead branch, and removes the reachable reflection code. The app then registers providers explicitly with `SqlAuthenticationProvider.SetProvider`.

### Also: `UseManagedNetworking` split

`UseManagedNetworking` mixed the OS check with the switch read, so the existing SNI substitution could only be embedded conditionally. Hoisting the platform guard out:

```csharp
public static bool UseManagedNetworking =>
    !OperatingSystem.IsWindows() || UseManagedNetworkingOnWindows;
```

leaves `UseManagedNetworkingOnWindows` as a pure substitution target. The guard makes the substitution safe on every platform, so `ILLink.Substitutions.xml` is now unconditional for .NET.

### Zero-code-change AOT is not achievable

One line of startup registration is required — the trimmer has no reason to keep `ActiveDirectoryAuthenticationProvider` without a static reference from the app. Rejected alternatives: `[ModuleInitializer]` in the extension (initializers of trimmed assemblies don't run — same chicken-and-egg), a source generator (viable but far more to ship and maintain), and a root descriptor XML (preserves the type but leaves the `Assembly.Load` path intact, which defeats the point). This matches the established .NET pattern, e.g. `JsonSerializerContext` registration under AOT.

### API changes

None. The switch is internal; the trimmer contract is the documented `RuntimeHostConfigurationOption` name.

## Issues

Fixes #4193 (together with #4670).

## Testing

- **New** `ILLinkSubstitutionsTests.cs` — asserts the substitution XML targets property getters that actually exist, so a rename can't silently break trimming.
- **Updated** `LocalAppContextSwitchesTest.cs` / `LocalAppContextSwitchesHelper.cs` for the new switch and the `s_useManagedNetworkingOnWindows` rename.
- **New** `tools/AotCompatibility` — publishes a NativeAOT app and inspects the output for `LoadAzureExtensionProvider`, confirming the feature switch removes it. See its README for expected output in both configurations.

Validation on this branch:

- `Microsoft.Data.SqlClient` + UnitTests build clean for `net9.0`, 0 warnings.
- Unit tests: 1188 passed, 14 skipped. The single failure (`SimulatedServerTests.ConnectionTests.IntegratedAuthConnectionTest`, "Cannot generate SSPI context") is pre-existing and environmental — no Kerberos on the Linux test host.

### Conflict resolutions worth reviewing

Replaying onto current `main` required resolving conflicts introduced by work that landed after the AOT branch point:

- **`LocalAppContextSwitches.cs`** — kept the AOT two-property design and the `s_useManagedNetworkingOnWindows` field; retained main's default-value documentation.
- **`LocalAppContextSwitchesHelper.cs`** — kept main's `GetLocalAppContextSwitchesType()` refactor and runtime OS check, adopting the AOT field rename.
- **`LocalAppContextSwitchesTest.cs`** — kept main's `switchesHelper` assertions (including `UseLegacyIdleTimeoutBehavior`) and added the new switch assertion.

## Stack

```
#4566 assembly-signing-sqlserver
  #4567 assembly-signing-logging
    #4568 assembly-signing-abstractions
      #4569 assembly-signing-sqlclient
        #4570 assembly-signing-azure
          #4670 auth-registry-abstractions
            THIS PR  aot-azure-fix
```

Previously this PR combined the registry relocation with the AOT work. The structural refactor is now split out into #4670 and this PR contains only the AOT changes. Supersedes the AOT content of #4348.

### CODE REVIEW

File: src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/AuthenticationProviderRegistry.cs [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    /// instance via the internal constructor to avoid mutating global state.
+    /// </remarks>
+    internal static AuthenticationProviderRegistry Instance { get; } = new();
+
+    /// <summary>
+    /// A registered provider together with whether it was registered as permanent (via
+    /// <see cref="SetPermanentProvider"/>) and therefore not overridable by <see cref="SetProvider"/>.
+    /// </summary>
+    /// <param name="Provider">The registered provider.  Never <see langword="null"/>.</param>
+    /// <param name="IsPermanent">Whether the provider must not be overridden by <see cref="SetProvider"/>.</param>
```
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/AuthenticationProviderRegistry.cs:55]: @paulmedynski "True if the provider ..."

File: src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/AuthenticationProviderRegistry.cs [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    /// </summary>
+    private readonly ConcurrentDictionary<SqlAuthenticationMethod, ProviderEntry> _providers = new();
+
+    #endregion
+
+    #region Construction
+
+    /// <summary>
+    /// Initializes a new, empty registry. Production code uses the shared <see cref="Instance"/>;
+    /// the constructor is exposed to tests so they can exercise registry behavior in isolation.
```
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/AuthenticationProviderRegistry.cs:71]: @paulmedynski "Instance.  This constructor is exposed so tests can exercise ..."

File: src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/AuthenticationProviderRegistry.cs [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+    {
+        return _providers.TryGetValue(authenticationMethod, out ProviderEntry entry)
+            ? entry.Provider
+            : null;
+    }
+
+    /// <summary>
+    /// Registers an overridable provider for the given authentication method.
+    /// </summary>
+    /// <returns>
```
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/AuthenticationProviderRegistry.cs:95]: @paulmedynski Missing param docs.

File: src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/AuthenticationProviderRegistry.cs [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    /// <exception cref="NotSupportedException">
+    /// The provider does not support the given authentication method.
+    /// </exception>
+    /// <exception cref="NullReferenceException">
+    /// <paramref name="provider"/> is <see langword="null"/>.
+    /// </exception>
+    internal bool SetProvider(SqlAuthenticationMethod authenticationMethod, SqlAuthenticationProvider provider)
+    {
+        if (!provider.IsSupported(authenticationMethod))
+        {
```
[the remark this answers is not in the document]
Comment (paulmedynski) [reply src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/AuthenticationProviderRegistry.cs:108]: No, we're in a nullable context and this is an internal method.  The public callers must ensure they abide by the API contract here.

File: src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/AuthenticationProviderRegistry.cs [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    /// <remarks>
+    /// Callers are responsible for verifying that the provider supports the authentication method
+    /// before registering it.
+    /// <para>
+    /// This is a last-in-wins operation: a later <see cref="SetPermanentProvider"/> call for the
+    /// same authentication method unconditionally replaces any previously registered provider
+    /// (permanent or not). Only <see cref="SetProvider"/> is blocked by an existing permanent
+    /// provider; <see cref="SetPermanentProvider"/> itself always overwrites.
+    /// </para>
+    /// </remarks>
```
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/AuthenticationProviderRegistry.cs:178]: @paulmedynski Missing param docs.

File: src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/SqlAuthenticationProvider.cs [hunk 6]
```diff
@@ -21,25 +21,17 @@ public virtual void BeforeUnload(SqlAuthenticationMethod authenticationMethod) {
...
-    //
-    // We would like to deprecate this method in favour of
-    // SqlAuthenticationProviderManager.SetProvider().
-    //
     public static bool SetProvider(
         SqlAuthenticationMethod authenticationMethod,
         SqlAuthenticationProvider provider)
     {
-        return Internal.SetProvider(authenticationMethod, provider);
+        return AuthenticationProviderRegistry.Instance.SetProvider(authenticationMethod, provider);
```
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/SqlAuthenticationProvider.cs:35]: @paulmedynski Think about enforcing not-null `provider` here.

File: src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/Strings.resx
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/Strings.resx:1]: @paulmedynski Should these resource strings live in a `resources/` sub- or sibling-directory?

File: src/Microsoft.Data.SqlClient.Extensions/Abstractions/test/SqlAuthenticationProviderTest.cs [hunk 7]
```diff
@@ -15,7 +20,8 @@ public class SqlAuthenticationProviderTest
     /// </summary>
     public SqlAuthenticationProviderTest()
```
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient.Extensions/Abstractions/test/SqlAuthenticationProviderTest.cs:21]: @paulmedynski Add a test for setting a null provider.

File: src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/SqlAuthenticationProvider.cs [hunk 8]
```diff
@@ -21,25 +21,17 @@ public virtual void BeforeUnload(SqlAuthenticationMethod authenticationMethod) {
...
     /// <include file='../doc/SqlAuthenticationProvider.xml' path='docs/members[@name="SqlAuthenticationProvider"]/GetProvider/*'/>
-    //
-    // We would like to deprecate this method in favour of
-    // SqlAuthenticationProviderManager.GetProvider().
-    //
     public static SqlAuthenticationProvider? GetProvider(
         SqlAuthenticationMethod authenticationMethod)
     {
-        return Internal.GetProvider(authenticationMethod);
+        return AuthenticationProviderRegistry.Instance.GetProvider(authenticationMethod);
```
Comment (mdaigle) [inline src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/SqlAuthenticationProvider.cs:27]: 🤖 **Bootstrap before reading or writing the shared registry**

`GetProvider` and `SetProvider` now access the initially empty registry without triggering discovery. If an application calls `SetProvider` before its first federated connection, `AuthenticationBootstrapper.LoadAzureExtensionProvider` later calls the overridable `Registry.SetProvider` path for every Entra method and replaces the application’s provider. Previously, entering `SqlAuthenticationProviderManager.SetProvider` initialized configuration and Azure defaults first, so the explicit provider remained the final override. The same change makes `GetProvider` return null before first authentication even when the Azure extension is available. Preserve the old initialization and precedence contract, either by bootstrapping before public registry access or by making default discovery add only missing providers.

File: src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/Strings.resx [hunk 9]
```diff
@@ -0,0 +1,123 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (mdaigle) [inline src/Microsoft.Data.SqlClient.Extensions/Abstractions/src/Strings.resx:1]: I don't think these files will be picked up by our localization flow, so any additions/changes won't receive updates. It only knows how to update the resx files at the existing location (https://sqlclientdrivers.visualstudio.com/ADO.Net/_git/Microsoft.Data.SqlClient?path=/Localize/LocProject.json). It looks like we can add an additional location? but it will take a bit of experimentation.

If you don't want to do that now, a cheap alternative without modifying the localization pipeline could be to copy the resource files over to this package at build time.

**Review by mdaigle — CHANGES_REQUESTED** [review]

File: src/Microsoft.Data.SqlClient.Extensions/Azure/test/AuthenticationBootstrapperGlobalTests.cs [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+
+namespace Microsoft.Data.SqlClient.Extensions.Azure.Test;
+
+/// <summary>
+/// Tests for the SqlClient-internal <c>AuthenticationBootstrapper</c> that run the full bootstrap
+/// and so mutate the process-wide <c>AuthenticationProviderRegistry.Instance</c>. They are
+/// serialized via the shared <c>SqlAuthenticationProvider</c> collection.
```
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient.Extensions/Azure/test/AuthenticationBootstrapperGlobalTests.cs:12]: @paulmedynski Stale collection name.

File: src/Microsoft.Data.SqlClient.Extensions/Azure/test/AuthenticationBootstrapperGlobalTests.cs [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+    // Forces the MDS bootstrapper to run by invoking its internal static Bootstrap() method via
+    // reflection. This project does not have InternalsVisibleTo from Microsoft.Data.SqlClient, so
+    // the method cannot be called directly.
+    //
+    // NOTE: This perturbs GLOBAL state -- Bootstrap() seeds the process-wide
+    // AuthenticationProviderRegistry.Instance (installing the Azure provider for the AD methods).
+    // That is why this class lives in the [Collection("SqlAuthenticationProviderGlobal")] collection,
+    // which serializes it with the other tests that mutate the shared registry.
+    //
+    // TODO(https://sqlclientdrivers.visualstudio.com/ADO.Net/_workitems/edit/41888):
```
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient.Extensions/Azure/test/AuthenticationBootstrapperGlobalTests.cs:98]: @paulmedynski I think this prerequisite work is done.

File: src/Microsoft.Data.SqlClient.Extensions/Azure/test/AuthenticationBootstrapperTests.cs [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+
+    // Invokes the MDS-internal AuthenticationBootstrapper.CreateAzureAuthenticationProvider via
+    // reflection, passing the REAL Azure provider and options types. Unwraps the reflection wrapper
+    // so a test observes the real exception type, if any.
+    //
+    // NOTE: This reflection is only needed because this project does not have InternalsVisibleTo
+    // from Microsoft.Data.SqlClient. This call has no global side effects -- it just returns a new
+    // provider instance.
+    //
+    // TODO(https://sqlclientdrivers.visualstudio.com/ADO.Net/_workitems/edit/41888):
```
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient.Extensions/Azure/test/AuthenticationBootstrapperTests.cs:115]: @paulmedynski I think we have IVT for this now.

File: src/Microsoft.Data.SqlClient.Extensions/Azure/test/WamBrokerTests.cs [hunk 13]
```diff
@@ -272,45 +271,4 @@ public void Ctor_Options_Null_ThrowsArgumentNullException()
         Assert.Throws<ArgumentNullException>(
             () => new ActiveDirectoryAuthenticationProvider((ActiveDirectoryAuthenticationProviderOptions)null!));
     }
-
```
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient.Extensions/Azure/test/WamBrokerTests.cs]: This coverage was split across several other tests:
- WAM-broker construction/configuration behavior is covered in:
  - `src/Microsoft.Data.SqlClient.Extensions/Azure/test/AuthenticationBootstrapperTests.cs`
  - These verify that `UseWamBroker = true` is passed to the options constructor and preserved on the resulting `ActiveDirectoryAuthenticationProvider`.
- Registration preserves the provider instance is covered in:
  - `src/Microsoft.Data.SqlClient.Extensions/Abstractions/test/AuthenticationProviderRegistryTest.cs`
  - `SetProvider()` followed by `GetProvider()` asserts the same instance is returned.
  - `SqlAuthenticationProviderTest.PublicApi_DelegatesToSharedInstance()` also verifies the public API and registry observe the same instance.
- Provider installation for all Active Directory methods is covered in:
  - `src/Microsoft.Data.SqlClient.Extensions/Azure/test/AuthenticationBootstrapperGlobalTests.cs`

File: aot-auth-provider-proposal.md
Comment (paulmedynski) [inline aot-auth-provider-proposal.md:1]: @paulmedynski Move this beside our other design docs.

File: src/Microsoft.Data.SqlClient/tests/UnitTests/Microsoft/Data/SqlClient/AuthenticationBootstrapperTests.cs [hunk 14]
```diff
@@ -0,0 +1,350 @@
...
+    {
+        // Precondition: confirm the Azure extension is not present in this test context, so the
+        // Azure-absent assumptions in these tests hold.
+        Assert.Throws<FileNotFoundException>(
+            () => Assembly.Load("Microsoft.Data.SqlClient.Extensions.Azure"));
+    }
+
+    // CreateAzureAuthenticationProvider tests ----------------------------------------------
+    //
+    // Each Stub* container mimics one shape the real Azure extension might expose:
```
Comment (paulmedynski) [inline src/Microsoft.Data.SqlClient/tests/UnitTests/Microsoft/Data/SqlClient/AuthenticationBootstrapperTests.cs:38]: @paulmedynski Do we need this now that we release all SqlClient family packages together?
